W-18543667 added new Generating API Documentation with Einstein jh (#305)
mulesoft/docs-exchange: modules/ROOT/pages/generating-documentation-with-ai.adoc

= Generating API Documentation With Einstein Generative AI
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images/

> Use Einstein generative AI to generate API documentation for REST APIs in Exchange, and streamline how your organization manages APIs. By using AI technologies, developers can simplify API documentation processes. AI creates API documentation that is consistent, accurate, and current. This frees developers to focus on strategic tasks and enhances developer satisfaction.

Reviewer: ```suggestion
Use Einstein generative AI to generate API documentation for REST APIs in Exchange and streamline how your organization manages APIs. By using AI, you can simplify your API documentation processes. Generative AI creates API documentation that is consistent, accurate, and current. This frees you to focus on strategic tasks and enhances developer satisfaction.
```

You can create documentation for single file REST APIs that don't have any documentation. From the asset details page in Exchange, generate the documentation, then manually edit and improve the documentation. After that, publish it to make it available to everyone in your organization.

This release supports:

> * REST APIs only

Reviewer: ```suggestion
* REST APIs
```

> * Single file format only

Reviewer: ```suggestion
* Single file format
```

+

> Multi-format files and fragments aren’t supported

Reviewer: ```suggestion
Multi-format files and fragments aren’t supported.
```

== Documentation Structure
AI-generated documentation in Anypoint Exchange standardizes API documentation, addressing inconsistency and incompleteness. The structure and format of the generated documentation is standardized across each API including fonts, headers, text size.

Generated documentation includes:

* Overview and purpose of the API
* Authentication requirements and security details
* Base URL and environments
* Detailed endpoint documentation including:

** Request and response examples 
** Query parameters
** Path parameters
** Request body schemas
** Response codes and examples
* Error handling and status codes
* Best practices for implementation
* Sample code snippets

== Before You Begin
Before generating API documentation with Einstein, you must have:

> * Salesforce Org or Trial account with Einstein enabled in Anypoint Platform.

Reviewer: ```suggestion
* Salesforce org or trial account with Einstein enabled in Anypoint Platform.
```

Reviewer: I'm not familiar with "trial account". Do you mean a trial Developer edition, or sandbox? https://developer.salesforce.com/free-trials

+
To enable Einstein in Anypoint Platform, see xref::access-management:enabling-einstein.adoc#enable-einstein-anypoint[Enable Einstein for Your Anypoint Platform Organization].

> * Einstein credits 

Reviewer: See slides 4 & 5 in this deck regarding how to document billing. https://docs.google.com/presentation/d/13DalsZak5ap-ie4mscsUli7rsVmo-qTnTHzsGNJNES0/edit?usp=sharing

Author: I shared this with my PM and Ipshita Nag. They said that it is Einstein credits are correct.

+

> This feature uses Salesforce Einstein Large Language Model (LLM).

Reviewer: I think we just use Large Language Model. I'd remove this sentence and replace with the Einstein Generative AI disclaimer.

https://confluence.internal.salesforce.com/spaces/DOCTEAM/pages/1128073452/Large+Language+Model+LLM+Einstein+Generative+AI

== Generate API Documentation with Einstein

> Generate documentation for REST APIs that don't have any documentation. You can edit and save a draft as many times as you like before publishing. Drafts aren't visible to others until they are published.

Reviewer: ```suggestion
Generate documentation for REST APIs that don't have any documentation. You can edit and save a draft as many times as you like before publishing. Drafts aren't visible to others until they're published.
```

> . Select the API from the catalog and click *Generate Documentation* from the asset details page.

Reviewer: Earlier you used "Generate documentation". Which capitalization is correct?

image:ex-generate-doc.png[]

> . Review the generated documentation in draft mode and discard to delete the draft, edit to make changes or publish it.

Reviewer: ```suggestion
. Review the generated documentation in draft mode. Discard to delete the draft, or edit or publish it.
```
